code/implementation/docker/common/src/main/java/io/cattle/platform/docker/transform/DockerTransformerImpl.java

```diff
@@ -388,6 +390,38 @@ void setLogConfig(Instance instance, Map<String, Object> fromInspect) {
 
         setField(instance, FIELD_LOG_CONFIG, logConfig);
     }
+    
+    @SuppressWarnings("unchecked")
+    void setUlimit(Instance instance, Map<String, Object> fromInspect) {
+        Object ulimits = CollectionUtils.getNestedValue(fromInspect, HOST_CONFIG, "Ulimits");
+
+        if (ulimits == null) {
+            return;
+        }
+
+        List<Ulimit> ret = new ArrayList<>();
+        if (ulimits instanceof List) {
+            for (Object ulimit : (List<Object>) ulimits) {
+                if (ulimit instanceof Map) {
+                    Ulimit l = new Ulimit();
+                    Map<String, Object> temp = (Map<String, Object>) ulimit;
+                    if (temp.get("Name") instanceof String) {
+                        l.setName(temp.get("Name").toString());
+                    }
+                    if (temp.get("Hard") instanceof Integer) {
```

Comment:
Can you use Number instead of integer here and below?
So, you'd check if it is an instance of Number and then cast it to a number and use Number's intValue() function to convert it to an int.